### PR TITLE
Broker Reconciliation occurs too frequently

### LINKED ISSUE #1513: controller_broker reconciles too frequently
The controller_broker reconciles approximately every 5 minutes under "normal" circumstances after the first full broker's relistDuration elapses regardless of the relistDuration.

shouldReconcileClusterServiceBroker() checks Now - LastTransitionTime  against the relistDuration, but when the status remains in a steady steady of Ready=True, the last transition time is not updated and thus we incorrectly indicate we should reconcile every time we check.

### PR DESCRIPTION
**Pull request by jboyd01**
fixes #1513 

Added a new field for Broker's last reconciled time.  When determining if we should reconcile the broker, check that the last reconciled time vs the Ready condition's LastTransitionTime as LTT won't change over time under normal good circumstances when status remains Ready=True.

### CODE REVIEW

File: pkg/apis/servicecatalog/types.go [hunk 1]
```diff
@@ -32,8 +32,9 @@ type ClusterServiceBroker struct {
 	metav1.TypeMeta
 	metav1.ObjectMeta
 
-	Spec   ClusterServiceBrokerSpec
-	Status ClusterServiceBrokerStatus
+	Spec           ClusterServiceBrokerSpec
+	Status         ClusterServiceBrokerStatus
+	LastReconciled metav1.Time
```
Comment (pmorie) [inline pkg/apis/servicecatalog/types.go]: This field should be part of status and should be called LastCatalogRetrievalTime, and should be a pointer (zero value time is not meaningful)

**Comment by staebler** [comment]
Would it make sense to update the Ready condition every time that the Controller successfully fetches the catalog from the broker? Logically, we are rebuilding the broker, so it doesn't seem inappropriate to me to change the transition time on the Ready condition.

**Comment by jboyd01** [comment]
@staebler  we considered updating the LastTransitionTime but that felt like it violated the semantics

**Comment by jboyd01** [comment]
@pmorie I've updated per your review

File: pkg/controller/controller_broker.go [hunk 2]
```diff
@@ -128,11 +128,14 @@ func shouldReconcileClusterServiceBroker(broker *v1beta1.ClusterServiceBroker, n
...
+				// RelistDuration since the last time we fetched the Catalog
 				duration := broker.Spec.RelistDuration.Duration
-				intervalPassed := now.After(condition.LastTransitionTime.Add(duration))
+				intervalPassed := true
+				if broker.Status.LastCatalogRetrievalTime != nil {
+					intervalPassed = now.After(broker.Status.LastCatalogRetrievalTime.Time.Add(duration))
+				}
 				if intervalPassed == false {
-					glog.V(10).Info(pcb.Message("Not processing because RelistDuration has not elapsed since the broker became ready"))
+					glog.V(10).Info(pcb.Message("Not processing because RelistDuration has not elapsed since the last relist."))
```
Comment (pmorie) [inline pkg/controller/controller_broker.go]: nit: no period

Comment (jboyd01) [reply pkg/controller/controller_broker.go]: ok, addressed

**Comment by kibbles-n-bytes** [comment]
@jboyd01 Why do you feel it violated the semantics? I aggree with @staebler; this is a new event that is being logged with a new `LastTransitionTime`. It just happens to be that the result of that event is the same as the previous successful fetching attempt.

**Comment by jboyd01** [comment]
@kibbles-n-bytes I'm viewing LastTransitionTime as transition from one state to another, ie Ready -> Not Ready...  I'd could be convinced otherwise, but in chatting with @pmorie he was in agreement.  Paul?

**Comment by staebler** [comment]
@jboyd01 You're right. I was using the scenario where the controller sets a Ready/False condition for a resource that already has a Ready/False condition. I thought that in that scenario the transition time was updated to the current time. I see now that the transition time is not changed.

**Comment by kibbles-n-bytes** [comment]
@jboyd01 Ah, I see. I thought the same as @staebler.

**Comment by kibbles-n-bytes** [comment]
@jboyd01 Can we get a test for the state where the relist duration has elapsed with respect to the `LastTransitionTime`, but not the `LastCatalogRetrievalTime`, and so the catalog is not relisted?

**Comment by jboyd01** [comment]
@kibbles-n-bytes Absolutely, updated tests should have been included.  I've updated adding an explicit test for:
 1.  LastTransitionTime a long time ago & LastCatalogRetrievalTime meets duration expiration (results in **relist**)
 2.  LastTransitionTime a long time ago & LastCatalogRetrievalTime fails to meet duration expiration (it was just relisted, **no relist**)

**Pull request merged**